File: saml-attribute-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__SamlAttributeProviderDescriptor.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.saml.SamlAttributeProvider;
import se.curity.identityserver.sdk.plugin.descriptor.SamlAttributeProviderPluginDescriptor;

public final class ${pluginName}SamlAttributeProviderDescriptor implements SamlAttributeProviderPluginDescriptor<${pluginName}SamlAttributeProviderConfig>
{
    @Override
    public Class<? extends SamlAttributeProvider> getSamlAttributeProvider()
    {
        return ${pluginName}SamlAttributeProvider.class;
    }

    @Override
    public String getPluginImplementationType()
    {
        return "${pluginName.toLowerCase()}";
    }

    @Override
    public Class<? extends ${pluginName}SamlAttributeProviderConfig> getConfigurationType()
    {
        return ${pluginName}SamlAttributeProviderConfig.class;
    }
}

File: kotlin-saml-attribute-provider-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__SamlAttributeProviderDescriptor.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.plugin.descriptor.SamlAttributeProviderPluginDescriptor

class ${pluginName}SamlAttributeProviderDescriptor: SamlAttributeProviderPluginDescriptor<${pluginName}SamlAttributeProviderConfig>
{
    override fun getSamlAttributeProvider() = ${pluginName}SamlAttributeProvider::class.java

    override fun getPluginImplementationType() = "${pluginName.toLowerCase()}"

    override fun getConfigurationType() = ${pluginName}SamlAttributeProviderConfig::class.java
}
